### PR TITLE
fix: redis sink handle complex data types

### PR DESCRIPTION
**Pull request by kevinbheda**
This solves #61

### CODE REVIEW

File: src/test/java/io/odpf/depot/redis/parsers/RedisHashSetEntryParserTest.java [hunk 1]
```diff
@@ -59,9 +60,38 @@ private void redisSinkSetup(String field) throws IOException {
         schemaKey = odpfMessageParser.getSchema(schemaKeyClass, descriptorsMap);
     }
```
Comment (prakharmathur82) [inline src/test/java/io/odpf/depot/redis/parsers/RedisHashSetEntryParserTest.java:62]: can we add more tests? timestamp? repeated? nonrepeated? 

what about error scenarios?

File: src/test/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessageTest.java [hunk 2]
```diff
@@ -325,6 +328,20 @@ public void shouldGetFieldByName() throws IOException {
         Assert.assertEquals("order-1", protoOdpfParsedMessage.getFieldByName("order_number", odpfMessageSchema));
     }
```
Comment (prakharmathur82) [inline src/test/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessageTest.java:330]: can we add more tests? non repeated but complex, nested, timestamp?

File: src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java [hunk 3]
```diff
@@ -24,6 +31,12 @@
 public class ProtoOdpfParsedMessage implements ParsedOdpfMessage {
     private final DynamicMessage dynamicMessage;
 
+    private JsonFormat.Printer jsonPrinter = JsonFormat.printer()
+            .omittingInsignificantWhitespace()
+            .preservingProtoFieldNames()
```
Comment (prakharmathur82) [inline src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java:36]: we should add explicit tests for all these three flags

**Review by prakharmathur82 — CHANGES_REQUESTED** [review]

File: src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java [hunk 4]
```diff
@@ -132,19 +145,22 @@ public Object getFieldByName(String name, OdpfMessageSchema odpfMessageSchema) {
...
-            Descriptors.FieldDescriptor descriptor = message.getDescriptorForType().findFieldByName(key);
-            if (descriptor == null) {
-                throw new IllegalArgumentException("Invalid field config : " + name);
-            }
-            currentValue = message.getField(descriptor);
+        String jsonString = "";
+        try {
+            jsonString = this.jsonPrinter.print(dynamicMessage);
+        } catch (InvalidProtocolBufferException e) {
+            throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (harikrishnakanchi) [reply src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java]: wrapped with DeserilizerException

File: src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java (hunk 3 above)
Comment (harikrishnakanchi) [reply src/main/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessage.java:40]: have added tests for preserving protofields option.

File: src/test/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessageTest.java (hunk 2 above)
Comment (harikrishnakanchi) [reply src/test/java/io/odpf/depot/message/proto/ProtoOdpfParsedMessageTest.java:332]: added tests for timestamp, duration, map. Repeated and nested fields already exist.

File: src/test/java/io/odpf/depot/redis/parsers/RedisHashSetEntryParserTest.java (hunk 1 above)
Comment (lavkesh) [reply src/test/java/io/odpf/depot/redis/parsers/RedisHashSetEntryParserTest.java:73]: added

**Review by lavkesh — APPROVED** [review]

**Review by lavkesh — APPROVED** [review]

**Pull request merged**